core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -41,9 +45,11 @@ private[history] abstract class ApplicationHistoryProvider {
    * Returns the Spark UI for a specific application.
    *
    * @param appId The application ID.
+   * @param attemptId The application attempt ID for apps with multiple attempts (or an empty
+   *                  string for apps with a single attempt).
    * @return The application's UI, or None if application is not found.
    */
-  def getAppUI(appId: String): Option[SparkUI]
+  def getAppUI(appId: String, attemptId: String): Option[SparkUI]
```

Comment:
any reason not to use an `Option[String]` for `attemptId` here (and everywhere else) as well?